### DISCUSSION TITLE
Report: Cloudflare & Authelia OIDC

### DISCUSSION THREAD

**Discussion by ja0nz** [authelia/authelia#4206] [Show and tell]
Hello👋
I am running a **Cloudflare -> Caddy -> Authelia** stack. While "basic auth" worked always smooth behind Cloudflare I had some hard time to figure out how to get Authelia OIDC up and running. The trick was in the end to simply exclude Authelia running at **auth.domain.tld** from proxification.

_CF DNS settings_
```
A     domain.tld 123.45.67.890 Proxied  Auto
CNAME auth       domain.tld    DNS only Auto
CNAME *          domain.tld    Proxied  Auto
```
I am just curious if somebody can shed some light on the WHY its like that. Or how Authelia may run behind cloudflare which I would rather prefer. I tried numerous switches but only this endpoint exclude 👆 helped.
Attached you'll find the **Response headers** from the proxied/unproxied requests.

`curlie https://auth.domain.tld/well-known/openid-configuration`

_proxied (non working)_
```
HTTP/2 200
date: Wed, 19 Oct 2022 12:09:59 GMT
content-type: application/json; charset=utf-8
content-length: 1414
alt-svc: h3=":443"; ma=86400, h3-29=":443"; ma=86400
cache-control: no-store
content-security-policy: default-src 'none';
permissions-policy: interest-cohort=()
pragma: no-cache
referrer-policy: strict-origin-when-cross-origin
vary: Accept-Encoding
x-content-type-options: nosniff
x-frame-options: SAMEORIGIN
x-xss-protection: 1; mode=block
cf-cache-status: DYNAMIC
report-to: {"endpoints":[{"url":"https:\/\/a.nel.cloudflare.com\/report\/v3?s=lNggTwMG6EYTUBuzAjsxJYXSQ7LcyN6X3px%2B0u730gLD3R%2B71Rc7chQCdVJnVZe555hzMP9v8lQ7yGlB7bQgu1Yz7I1Rt7WIdC7tIium%2BnYNx9qgw3sEIl2mjVOTd8v3i6nHC0Pge3CXd1R8NSHNqQ%3D%3D"}],"group":"cf-nel","max_age":604800}
nel: {"success_fraction":0,"report_to":"cf-nel","max_age":604800}
strict-transport-security: max-age=15552000; includeSubDomains; preload
server: cloudflare
cf-ray: 75c9686fabf0b3a9-MUC
```
_unproxied/DNS only (working)_
```
HTTP/2 200
alt-svc: h3=":443"; ma=2592000
cache-control: no-store
content-security-policy: default-src 'none';
content-type: application/json; charset=utf-8
date: Wed, 19 Oct 2022 12:29:43 GMT
permissions-policy: interest-cohort=()
pragma: no-cache
referrer-policy: strict-origin-when-cross-origin
server: Caddy
vary: Accept-Encoding
x-content-type-options: nosniff
x-frame-options: SAMEORIGIN
x-xss-protection: 1; mode=block
content-length: 1414
```

**Comment by james-d-elliott** [comment]
I run it fine with it proxied via Cloudflare. You may want to elaborate. The content length for both responses was the same also, which is a good indication of nothing changing.

**Reply by ja0nz** [reply]
Ahh, I was testing against the OpenID Debugger https://openidconnect.net/
The problem was that **Cloudflares Bot Protection** was blocking the requests. After turning this feature off it runs fine.

Still I am experiencing issues with my WikiJS setup. The auth flow flows as expected until the callback GET request which times out (Code 524). This is not an Authelia issue but maybe you have an idea. Anyway, thanks for your reply and I'll try to get it running🤞
